rdrf/rdrf/email_notification.py

```diff
@@ -17,55 +16,111 @@ class RdrfEmail(object):
 
     _DEFAULT_LANGUAGE = "en"
 
-    def __init__(self, reg_code=None, description=None, language="en", email_notification=None):
+    def __init__(self, reg_code=None, description=None, email_notification=None, language=None):
         self.email_from = None
-        self.recipient = []
+        self.recipients = []
         self.email_templates = []
-        self.language = language
         self.template_data = {}
 
         self.reg_code = reg_code
         self.description = description
+        self.language = language # used to only send to subset of languages by EmailNotificationHistory resend
 
-        self.email_notification = email_notification
+        if email_notification:
+            self.email_notification = email_notification
+            self.reg_code = self.email_notification.registry.code
+            self.description = self.email_notification.description
+            
+        else:
+            self.email_notification = self._get_email_notification()
+            
 
     def send(self):
```

Comment:
Maybe it is just me, but this `RdrfEmail` abstraction got a bit messy for me.
It looked to me like it represented just one instance of an email we send out, but now from the `send` we can send out multiple emails. There is also some skipping at line 51, and comments says resend will happen per language template which isn't clear how it happens.

Can't we just at the notification level: 
  * prepare all the emails that have to be sent out
  * iterate through that list and create one `RdrfEmail` for each email and send it out.

That will separate concerns a bit better and will keep the `RdrfEmail.send()` simple as it was.